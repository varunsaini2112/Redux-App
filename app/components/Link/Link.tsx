import { Pressable, Text } from "react-native";

import { LinkProps } from "../../types";
import { fontSizes, fontStyles } from "../../config/appStyles";

const Link = ({ value, onPress }: LinkProps) => {
	return (
		<Pressable onPress={onPress}>
			<Text style={[fontStyles.content, { fontSize: fontSizes.xl }]}>
				{value}
			</Text>
		</Pressable>
	);
};

export default Link;
