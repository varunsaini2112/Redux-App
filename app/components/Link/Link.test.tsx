import { fireEvent, render, screen } from "@testing-library/react-native";
import Link from "./Link";

const mockOnPress = jest.fn();

describe("Link Component", () => {
	it("should match snapshots", () => {
		const snapshot = render(<Link value="Dummy link" onPress={mockOnPress} />);

		expect(snapshot).toMatchSnapshot();
	});

	it("onPress should work correctly", () => {
		render(<Link value="Dummy link" onPress={mockOnPress} />);

		fireEvent.press(screen.getByText("Dummy link"));

		expect(mockOnPress).toHaveBeenCalled();
	});
});
