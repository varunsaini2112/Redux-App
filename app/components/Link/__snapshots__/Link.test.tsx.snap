// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component should match snapshots 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      [
        {
          "color": "#999999",
          "fontSize": 14,
          "fontWeight": "500",
        },
        {
          "fontSize": 20,
        },
      ]
    }
  >
    Dummy link
  </Text>
</View>
`;
